perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ConsentsManagerBlImpl.java

```diff
@@ -38,6 +45,28 @@ public void setPerunBl(PerunBl perunBl) {
 		this.perunBl = perunBl;
 	}
 
+	public Consent createConsent(PerunSession sess, Consent consent) throws ConsentExistsException, UserNotExistsException, PrivilegeException, ConsentHubNotExistsException, ConsentNotExistsException {
```

Comment:
When creating a consent, it should always contain all attribute definitions required in the consent hub (all user-related attributes required on services assigned to resources of the consent hub's facilities). Therefore, we should always calculate these attributes and set them to the consent object before it is passed to the impl layer, where the attributes are inserted into the `consent_attr_defs` table.

Please, mention this (ignoring attributes in the consent object and calculating them instead) in the method's Javadoc because it is not a typical "create" behavior.